### DISCUSSION TITLE
Random values on wire (out of options)

### DISCUSSION THREAD

**Discussion by User32sys** [hneemann/Digital#1311] [Q&A] [answered]
I have created endless designs to get rid of this problem but with each design it creeps back in. Definitely need som help here. Searched in the discussions and issues and read about random values of the driver when an input is high-z. Tried to eliminate that but to no avail. When stepping through my design with the clock, at step 12 there should be a value of 7 in the Dout of the IR register. 
There is something i am missing but i have nu clue. Help is greatly appreciated.
[CPU0.zip](https://github.com/user-attachments/files/16513107/CPU0.zip)

**Comment by hneemann** [comment, marked as answer]
The simulator behaves similarly to a real CMOS circuit: If an input is set to high-z, the value read is more or less random.
If you want to avoid this, you can use a pull-up or pull-down resistor to pull the high-z value to high or low.

**Reply by User32sys** [reply]
Yes, that was missing. Thank you very much.
